addons/api/mirage/config.js

```diff
@@ -464,19 +464,31 @@ export default function () {
       // If adding credential sources, push them into the array
       if (method === 'add-credential-sources') {
         updatedAttrs.credentialLibraryIds = target.credentialLibraryIds;
+        updatedAttrs.credentialIds = target.credentialIds;
         attrs.applicationCredentialSourceIds.forEach((id) => {
-          if (!updatedAttrs.credentialLibraryIds.includes(id)) {
-            updatedAttrs.credentialLibraryIds.push(id);
+          if (
+            !updatedAttrs.credentialLibraryIds.includes(id) ||
+            !updatedAttrs.credentialIds.includes(id)
+          ) {
+            if (id.includes('cred')) {
```

Comment:
Is it safe to assume that other id's won't have `cred` as part of them?

Reply:
For now it is safe. Static `username_password` credentials are prefixed with `cred_` while `vault` credential libraries are prefixed with `cl_`. I am fairly certain that this will need another refactor for future credential injection work.

Reply:
You might use `startsWith` to be more specific.

Reply:
Good suggestion. Fixed!